### ISSUE TITLE
config.warningsAsErrors not behaving as expected.

### ISSUE THREAD

**Issue by sheinbergon** [detekt/detekt#3311]
I'm probably missing something, but I'm failing to get the detekt gradle plugin to fail my build.  My yaml configuration files starts with 

```yaml
build:
  maxIssues: 1
  weights:

config:
  validation: true
  warningsAsErrors: true

```

and an error such as `ModifierOrder` is reported, but the build succeeds

## Expected Behavior
A rule with a default severity level of warning should fail the task

## Observed Behavior
The build succeeds while warning about the order

## Steps to Reproduce
- Make sure the `ModifierOrder` rule is active
- Create a line violating the proper modifier order (`lateinit override var field:Int`)
- Run the build

## Environment
- JDK 11.0.8
- Kotlin 1.4.20
- Gradle 6.7.1
- Detekt version 1.14.1 and 1.15.0
